engine/suppliers.go

```diff
@@ -302,12 +306,118 @@ func (spS *SupplierService) sortedSuppliersForEvent(args *ArgsGetSuppliers) (sor
 }
 
 type ArgsGetSuppliers struct {
-	IgnoreError bool
-	MaxCost     float64
+	IgnoreErrors bool
+	MaxCost      string
 	utils.CGREvent
 	utils.Paginator
 }
 
+func (args *ArgsGetSuppliers) asOptsGetSuppliers(supp []*Supplier) (out *optsGetSuppliers, err error) {
+	out = new(optsGetSuppliers)
+	if args.MaxCost != "" {
+		if args.MaxCost == utils.MetaEventCost {
+			val, err := out.costForEvent(&args.CGREvent, supp)
+			if err != nil {
+				return nil, err
+			}
+			out.maxCost = val
+		} else {
+			val, err := strconv.ParseFloat(args.MaxCost, 64)
+			if err != nil {
+				return nil, err
+			}
+			out.maxCost = val
+		}
+	}
+	out.ignoreErrors = args.IgnoreErrors
+	return
+}
+
+type optsGetSuppliers struct {
+	ignoreErrors bool
+	maxCost      float64
+}
+
+func (args *optsGetSuppliers) costForEvent(ev *utils.CGREvent, suppls []*Supplier) (val float64, err error) {
```

Comment:
Here I find a bit overloaded to create costForEvent on opts. Maybe we can make a simple function doing it, so we can reuse the code later.